funcbench/env.go

```diff
@@ -0,0 +1,224 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/github"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+)
+
+type Environment interface {
+	BenchFunc() string
+	CompareTarget() string
+	IsRaceEnabled() bool
+
+	PostErr(err string) error
+	PostResults(results string) error
+}
+
+type environment struct {
+	logger Logger
+
+	benchFunc     string
+	compareTarget string
+	isRaceEnabled bool
+
+	home string
+}
+
+func (e environment) BenchFunc() string     { return e.benchFunc }
+func (e environment) CompareTarget() string { return e.compareTarget }
+func (e environment) IsRaceEnabled() bool   { return e.isRaceEnabled }
+
+type Local struct {
+	environment
+}
+
+func newLocalEnv(e environment) (Environment, error) {
+	return &Local{environment: e}, nil
+
+}
+
+func (l Local) PostErr(string) error { return nil } // Noop. We will see error anyway.
+
+func (l Local) PostResults(results string) error {
+	fmt.Println("Results:")
+	fmt.Println(results)
+
+	// Also save to file.
+	if err := ioutil.WriteFile(time.Now().String(), []byte(results), os.ModePerm); err != nil {
+		return err
+	}
+	return nil
+}
+
+// TODO: Add unit test.
+type GitHub struct {
+	environment
+
+	client  *gitHubClient
+	logLink string
+}
+
+func newGitHubEnv(logger Logger, eventFilePath string, c *commander) (Environment, error) {
+	data, err := ioutil.ReadFile(eventFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	event, err := github.ParseWebHook("issue_comment", data)
+	if err != nil {
+		return nil, err
+	}
+
+	email, err := c.exec("git", "config", "--get", "user.email")
+	if err != nil {
+		return nil, err
+	}
+
+	if email != "" {
+		return nil, errors.Errorf("expected fresh machine, got git user configured: %s", email)
```

Comment:
why is this check?

Reply:
To not mess up your own git config - I am running this locally.